src/utils/parsers.js

```diff
@@ -0,0 +1,27 @@
+// 커스텀 구분자 추출
+export function extractCustomDelimiter(input) {
+  const CUSTOM_DELIMITER_REGEX = /^\/\/(.+?)\\n/;
+  const customDelimiterMatch = input.match(CUSTOM_DELIMITER_REGEX);
+  return customDelimiterMatch ? customDelimiterMatch[1] : null;
+}
+
+// 숫자 문자열 추출
+export function getNumberString(input, customDelimiter) {
+  const CUSTOM_DELIMITER_REGEX = /^\/\/(.+?)\\n/;
+  const customDelimiterLength = customDelimiter
+    ? input.match(CUSTOM_DELIMITER_REGEX)[0].length
+    : 0;
+  return input.slice(customDelimiterLength).trim();
+}
+
```

Comment:
위의 두 함수가 같은 CUSTOM_DELIMITER_REGEX를 가지고 match하는 코드인데 따로 분리하신 이유가 있나요?
저는 똑같은 상황에 한 함수에 같이 구하고 return했어서, 분리한 의도를 배워가고 싶습니다.

Reply:
extractCustomDelimiter -> 사용자가 입력한 커스텀 구분자를 추출하기 위해 작성한 함수
getNumberString -> 커스텀 구분자가 있다면 커스텀 구분자 부분(ex. //;\n)을 slice하여 반환 || 없으면 0을 slice하여 입력한 숫자 부분(ex. 3,4,5) 반환

두 가지 기능이 다르기 때문에 분리했습니다! CUSTOM_DELIMITER_REGEX 상수는 전역에 선언해서 사용할 수 있게 했으면 더 효율적이었을 것 같은데, 제가 놓친 것 같습니다!

Reply:
이 부분에서 input.match(CUSTOM_DELIMITER_REGEX)[0].length 가 모종의 이유로 매칭에 실패했을 경우 null 값이 나올 수도 있기 때문에 조금 더 방어적으로 작성해보는건 어떨까요

저라면 이렇게 작성했을 것 같아요

```suggestion

// 커스텀 구분자의 길이를 계산하는 함수
// 추후 커스텀 구분자의 선언부 형태가 달라질 수 있기 때문에 regex 형태로 원하는 형태를 입력받음
export function getCustomDelimiterLength(input, regex) {
  const match = input.match(regex);
  return match ? match[0].length : 0;
}

// 숫자 부분을 반환하는 함수
export function getNumberString(input, customDelimiter, regex) {
  let customDelimiterLength = 0;

  // 커스텀 구분자가 있을 경우 길이를 계산
  if (customDelimiter) {
    customDelimiterLength = getCustomDelimiterLength(input, regex);
  }

  // 구분자 길이만큼 자른 후 문자열 반환
  return input.slice(customDelimiterLength).trim();
}
```